src/App.jsx

```diff
@@ -14,45 +17,99 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_BY_ALPHABET = 'alphabetically';
+const SORT_BY_LENGTH = 'length';
+
+function getPreparedGoods(goods, sortField, reverse) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_BY_ALPHABET:
+          return good1.localeCompare(good2);
+
+        case SORT_BY_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    preparedGoods.reverse();
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverse, setreverse] = useState(false);
+  const visibleGoods = getPreparedGoods(
+    goodsFromServer,
+    sortField,
+    reverse,
+  );
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          onClick={() => (
+            setSortField(SORT_BY_ALPHABET))}
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== SORT_BY_ALPHABET,
+          })}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortField !== SORT_BY_LENGTH,
+          })}
+          onClick={() => (
+            setSortField(SORT_BY_LENGTH)
+          )}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': !reverse,
+          })}
+          onClick={() => setreverse(!reverse)}
+        >
+          Reverse
+        </button>
+
+        {(sortField || reverse)
+        && (
+        <button
+          type="button"
+          className="button is-danger is-light"
+          onClick={() => {
+            setSortField('');
+            setreverse(false);
+          }}
```

Comment:
It's better to create separate functions to handle events. This way, your code will be easier to maintain and change
```suggestion
          onClick={onReset}
```